src/datadog/tracer_config.h

```diff
@@ -176,6 +178,79 @@ struct TracerDefaultConfig final {
   int remote_configuration_poll_interval_seconds = 5;
 };
 
+enum class ConfigName : char {
+  SERVICE_NAME,
+  SERVICE_ENV,
+  SERVICE_VERSION,
+  REPORT_TRACES,
+  TAGS,
+  TRACE_SAMPLING_RATE,
+  EXTRACTION_STYLES,
+  INJECTION_STYLES,
+  STARTUP_LOGS,
+  REPORT_TELEMETRY,
+  DELEGATE_SAMPLING,
+  GENEREATE_128BIT_TRACE_IDS,
+  AGENT_URL,
+  RC_POLL_INTERVAL,
+  TRACE_SAMPLING_LIMIT,
+  TRACE_SAMPLING_RULES,
+  SPAN_SAMPLING_RULES,
+};
+
+inline std::string to_string(ConfigName name) {
+  switch (name) {
+    case ConfigName::SERVICE_NAME:
+      return "DD_SERVICE";
+    case ConfigName::SERVICE_ENV:
+      return "DD_ENV";
+    case ConfigName::REPORT_TRACES:
+      return "trace_enabled";
+    case ConfigName::TAGS:
+      return "trace_tags";
+    case ConfigName::TRACE_SAMPLING_RATE:
+      return "trace_sample_rate";
+    case ConfigName::SERVICE_VERSION:
+      return "DD_VERSION";
+    case ConfigName::EXTRACTION_STYLES:
+      return "DD_TRACE_PROPAGATION_STYLE_EXTRACT";
+    case ConfigName::INJECTION_STYLES:
+      return "DD_TRACE_PROPAGATION_STYLE_INJECT";
+    case ConfigName::STARTUP_LOGS:
+      return "DD_TRACE_STARTUP_LOGS";
+    case ConfigName::REPORT_TELEMETRY:
+      return "DD_INSTRUMENTATION_TELEMETRY_ENABLED";
+    case ConfigName::DELEGATE_SAMPLING:
+      return "DD_TRACE_DELEGATE_SAMPLING";
+    case ConfigName::GENEREATE_128BIT_TRACE_IDS:
+      return "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED";
+    case ConfigName::AGENT_URL:
+      return "DD_TRACE_AGENT_URL";
+    case ConfigName::RC_POLL_INTERVAL:
+      return "DD_REMOTE_CONFIG_POLL_INTERVAL_SECONDS";
+    case ConfigName::TRACE_SAMPLING_LIMIT:
+      return "DD_TRACE_RATE_LIMIT";
+    case ConfigName::SPAN_SAMPLING_RULES:
+      return "DD_SPAN_SAMPLING_RULES";
+    case ConfigName::TRACE_SAMPLING_RULES:
+      return "DD_TRACE_SAMPLING_RULES";
+  }
+
+  std::abort();
+}
```

Comment:
that's telemetry name, maybe move this conversion method to `tracer_telemetry.cpp`?